unit/ctest_rect_grid.c

```diff
@@ -35,6 +35,139 @@ void test_grid_2d()
     }
 }
 
+void test_find_cell_1x(){
```

Comment:
please add some (very brief) comments to your testing functions to at least say which case you are testing in each bit of code. So future readers know the purpose of each bit.